cmd/replication/main.go

```diff
@@ -40,7 +41,23 @@ func main() {
 	var wg sync.WaitGroup
 	doneC := make(chan bool, 1)
 	tracing.GoPanicWrap(ctx, &wg, "main", func(ctx context.Context) {
-		s, err := server.NewReplicationServer(ctx, log, options, registry.NewFixedNodeRegistry([]registry.Node{}))
+		db, err := db.NewDB(
+			ctx,
+			options.DB.WriterConnectionString,
+			options.DB.WaitForDB,
+			options.DB.ReadTimeout,
+		)
+		if err != nil {
+			log.Fatal("initializing database", zap.Error(err))
+		}
+
+		s, err := server.NewReplicationServer(
+			ctx,
+			log,
+			options,
+			registry.NewFixedNodeRegistry([]registry.Node{}),
+			db,
+		)
```

Comment:
Pulled the DB out of the replication server for dependency injection purposes. Otherwise, multiple instantiated servers in a test would share the same database and/or have cleanup issues.

Reply:
That makes sense